src/Cake.Common/Tools/GitVersion/GitVersionRunner.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+using Cake.Core.IO;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.GitVersion
+{
+    /// <summary>
+    /// The GitVersion runner.
+    /// </summary>
+    public sealed class GitVersionRunner : Tool<GitVersionSettings>
+    {
+        private ICakeLog _log;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GitVersionRunner"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="globber">The globber.</param>
+        /// <param name="log">The log.</param>
+        public GitVersionRunner(IFileSystem fileSystem, ICakeEnvironment environment, IProcessRunner processRunner,
+            IGlobber globber, ICakeLog log)
+            : base(fileSystem, environment, processRunner, globber)
+        {
+            _log = log;
+        }
+
+        /// <summary>
+        /// Runs GitVersion and processes the results.
+        /// </summary>
+        /// <param name="settings">The settings.</param>
+        /// <returns>A task with the GitVersion results.</returns>
+        public GitVersion Run(GitVersionSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            if (settings.OutputType != GitVersionOutput.BuildServer)
+            {
+                var jsonString = string.Empty;
+
+                Run(settings, GetArguments(settings), new ProcessSettings { RedirectStandardOutput = true },
+                process => jsonString = string.Join("\n", process.GetStandardOutput()));
+
+                var deserializer = new DataContractJsonSerializer(typeof(GitVersion));
+                return deserializer.ReadObject(new MemoryStream(Encoding.UTF8.GetBytes(jsonString))) as GitVersion;
```

Comment:
Any special reason the `MemoryStream` isn't disposed/within a using statement here?